### PR TITLE
[backport] Fix `F.softmax` and `F.log_softmax` with `axis=-1` on gpu

### PR DESCRIPTION
**Pull request by kmaehashi**
Backport of #5496

### CODE REVIEW

**Comment by kmaehashi** [comment]
[automatic post] Jenkins, test this please.
